tracing-opentelemetry-http/src/main/java/io/micronaut/tracing/opentelemetry/instrument/http/server/MicronautHttpServerAttributesGetter.java

```diff
@@ -103,8 +103,9 @@ public String target(HttpRequest<Object> request) {
     @Override
     @Nullable
     public String route(HttpRequest<Object> request) {
-        Optional<Object> routeInfo = request.getAttribute(HttpAttributes.ROUTE_INFO);
-        return routeInfo.map(Object::toString).orElse(null);
+        String route = request.getAttribute(HttpAttributes.URI_TEMPLATE).map(Object::toString)
+            .orElse(request.getUri().toString());
+        return request.getMethodName() + " - " + route;
```

Comment:
For unmatched routes, either ROUTE_INFO or URI_TEMPLATE would return null. I added a fall back using the raw route so at least those 404 not found are still reported with correct paths. 

Looks like no tests needed to be updated. I'm not too familiar with groovy tests also lol.

Reply:
Hi @pixnbit, This makes sense. I think we should also change

`statusCode` method to something like this:

```
    @Override
    public Integer statusCode(HttpRequest<Object> request, HttpResponse<Object> response) {
        if (response.code() >= 400) {
            Span.current().setStatus(StatusCode.ERROR);
        }
        return response.code();
    }
```

And it would be great to add one test in `OpenTelemetryHttpSpec` something that would looks like:
```
    void 'test error 404'() {
        when:
        def route = '/error/notFoundRoute'
        HttpResponse<String> response = reactorHttpClient.toBlocking().exchange(route, String)

        then:
        def e = thrown(HttpClientResponseException)
        e.message == "Not Found"
        conditions.eventually {
            exporter.finishedSpanItems.size() == 1
            exporter.finishedSpanItems[0].name == "GET - " + route
            exporter.finishedSpanItems[0].status.statusCode == StatusCode.ERROR
        }
        cleanup:
        exporter.reset()
    }
  ```

Reply:
Sure, I'll update.

Reply:
I just updated, literally just pasted your code.
But it feels strange setting the status code on a getter.

Reply:
I think you are right its better to be done inside `OpenTelemetryServerFilter` class inside `createOpenTelemetryPublisher`. It can be done something like:

``` 
           @Override
            public void doOnNext(@NonNull MutableHttpResponse<?> object, @NonNull Context context) {
                OpenTelemetryObserver.super.doOnNext(object, context);
                if (object.status().getCode() >= 400 ){
                        Span.current().setStatus(StatusCode.ERROR);
                }
            }
```

Reply:
Updated! This is better